### PR TITLE
Fix flaky test api4/Test_applyIPFilters

### PR DESCRIPTION
**Pull request by nickmisasi**
#### Summary
Master pipeline has a flaky test - it seems that there is some propagation issue with both the license and the cloud mock, and their usage inside a go routine. See thread here for more discussion on the issue: https://community.mattermost.com/private-core/pl/taeawck8up8cpqesimi5sro96y

For now I've opted to move the license and cloud check outside of the go routine, resolving the flakiness. Will follow up later with the server team to try and figure out the root cause of the propagation issues

#### Ticket Link
https://mattermost.atlassian.net/browse/CLD-6627

#### Release Note

```release-note
None
```

### CODE REVIEW

File: server/channels/api4/ip_filtering.go [hunk 1]
```diff
@@ -83,7 +83,18 @@ func applyIPFilters(c *Context, w http.ResponseWriter, r *http.Request) {
 
 	auditRec.Success()
 
-	c.App.Srv().Go(func() {
+	cloudWorkspaceOwnerEmailAddress := ""
```
Comment (lieut-data) [inline server/channels/api4/ip_filtering.go:86]: nit: this "feels" like something that might belong in the app layer vs. directly in the api4 endpoint. Not blocking :)

**Review by lieut-data — APPROVED** [review]

**Review by gabrieljackson — APPROVED** [review]
LGTM

File: server/channels/api4/ip_filtering.go (hunk 1 above)
Comment (nickmisasi) [reply server/channels/api4/ip_filtering.go:86]: I've opened a ticket here: https://mattermost.atlassian.net/browse/CLD-6630 to address this in a future PR

**Pull request merged**